preview_generator/preview/builder/image__pillow.py

```diff
@@ -1,16 +1,193 @@
+            self,
+            origin_image: PIL.Image,
+            file_output: BytesIO,
+            optimize: bool = DEFAULT_JPEG_OPTIMIZE,
+            quality: int = DEFAULT_JPEG_QUALITY,
+            progressive: bool = DEFAULT_JPEG_PROGRESSIVE,
+            saving_mode: str = DEFAULT_SAVING_MODE,
+            background_color: typing.Union[int, typing.Tuple[int, int, int]] = DEFAULT_BW_BACKGROUND_COLOR
+    ) -> BytesIO:
+        # INFO - G.M - 2019-06-25 - create uniform base image
+        temp_image = Image.new(
+            mode=saving_mode,
+            size=(origin_image.width, origin_image.height),
+            color=background_color,
+        )
+        # INFO - G.M - 2019-06-25 - apply current image with transparency on top of the base image
+        try:
+            temp_image.paste(im=origin_image, box=(0, 0), mask=origin_image)
+        except ValueError:
+            # self.logger.warning(
+            #     'Failed the transparency mask superposition. '
+            #     'Maybe your image does not contain a transparency mask')
+            temp_image.paste(origin_image)
+        temp_image.save(fp=file_output, format='jpeg', optimize=optimize, quality=quality,
+                          progressive=progressive)
+        file_output.seek(0, 0)
+        return file_output
+
+class RGBAImageConvertStrategy(TransparentImageConvertStrategy):
+    """
+    Image converter for RGBA image
+    """
+
+    def save(
+            self,
+            origin_image: PIL.Image,
+            file_output: BytesIO,
+            optimize: bool = DEFAULT_JPEG_OPTIMIZE,
+            quality: int = DEFAULT_JPEG_QUALITY,
+            progressive: bool= DEFAULT_JPEG_PROGRESSIVE
+    ) -> BytesIO:
+        return self.save_transparency(
+            origin_image=origin_image,
+            file_output=file_output,
+            optimize=optimize,
+            quality=quality,
+            progressive=progressive,
+            saving_mode='RGB',
+            background_color=DEFAULT_RGB_BACKGROUND_COLOR,
+        )
+
+class LAImageConvertStrategy(TransparentImageConvertStrategy):
+    """
+    Image converter for LA image
+    """
+    def save(
+            self,
+            origin_image: PIL.Image,
+            file_output: BytesIO,
+            optimize: bool = DEFAULT_JPEG_OPTIMIZE,
+            quality: int = DEFAULT_JPEG_QUALITY,
+            progressive: bool= DEFAULT_JPEG_PROGRESSIVE
+    ) -> BytesIO:
+        return self.save_transparency(
+            origin_image=origin_image,
+            file_output=file_output,
+            optimize=optimize,
+            quality=quality,
+            progressive=progressive,
+            saving_mode='L',
+            background_color=DEFAULT_BW_BACKGROUND_COLOR,
+        )
+
+class PillowImageConvertStrategyFactory(object):
+    """
+    Factory class to retrieve image convert strategy according to image mode
+    """
+
+    def get_strategy(self, mode: str) -> ImageConvertStrategy:
```

Comment:
mode should be an enum, not a raw string. otherwith, the user of the class must know available values

Reply:
Problem is that we rely on Pillow code and pillow don't give access to such enum. If tomorrow, pillow support a new mode, code should still work.

Reply:
does it make sense to create enum for something that doesn't rely on our code ?

Reply:
probably the best way to fix this is to use the image as parameter instead of the image  (and the factory code is `if image.mode == 'RGBA': [...]`)

Does it make sense to use the factory without an image? Maybe this is the right way to do it.

Reply:
done.